Products/ZenCollector/scheduler.py

```diff
@@ -154,7 +157,14 @@ def late(self):
         """
         Called whenever this task is late and missed its scheduled run time.
         """
-        # TODO: report an event
+        try:
+            # send event only for missed runs on devices.
+            self.task._eventService.sendEvent(
+                STATUS_EVENT, device=self.task._devId,
+                summary="Missed run on {}".format(self.task._manageIp),
+                component="zencommand", severity=Event.Warning)
```

Comment:
That has nothing to do with the new events you are sending.  Those events are only for zencommand and are specific events sent by the zencommand task.  The code you added will send events for all collectors.  Additionally your have since added the event class of Cmd_Fail which is probably not the correct class to use.